### PR TITLE
Updating spglib attr access / improvements for MP building

### PR DESCRIPTION
**Pull request by esoteric-ephemera**
Smaller changes to robocrys:
- Changing how spglib dataset attrs are accessed (see [this PMG PR](https://github.com/materialsproject/pymatgen/pull/3923))
- More flexibility with loading mineral fingerprint dataset (cc @tsmathis) to avoid unnecessary reloading during emmet building of robocrys docs

### CODE REVIEW

File: robocrys/condense/component.py [hunk 1]
```diff
@@ -183,7 +183,10 @@ def get_sym_inequiv_components(
     components = deepcopy(components)
 
     sym_inequiv_components = {}
-    equivalent_atoms = spg_analyzer.get_symmetry_dataset()["equivalent_atoms"]
+    if hasattr(symm_dset := spg_analyzer.get_symmetry_dataset(),"equivalent_atoms"):
```
Comment (janosh) [inline robocrys/condense/component.py:186]: alternatively, you could pin `spglib>=2.5` in `setup.py` and only support the attr version

Comment (esoteric-ephemera) [reply robocrys/condense/component.py:186]: No particular preference from my side, maybe it's better to be more flexible wrt spglib version? Happy to go with your rec

Comment (tsmathis) [reply robocrys/condense/component.py:186]: I would also lean towards Janosh's rec on this on one

**Comment by esoteric-ephemera** [comment]
Should be ready to merge whenever you get a chance!

**Review by janosh — APPROVED** [review]
thanks @esoteric-ephemera! 👍

**Pull request merged**